pkg/apis/serving/metadata_validation.go

```diff
@@ -63,18 +63,16 @@ func validateKnativeAnnotations(annotations map[string]string) (errs *apis.Field
 	return
 }
 
-const noAutoscalingMsg = "annotations in group 'autoscaling.knative.dev' won't work here, they must be put under 'spec.template.metadata.annotations' instead"
-
 // ValidateHasNoAutoscalingAnnotation validates that the respective entity does not have
 // annotations from the autoscaling group. It's to be used to validate Service and
 // Configuration.
-func ValidateHasNoAutoscalingAnnotation(annotations map[string]string) *apis.FieldError {
+func ValidateHasNoAutoscalingAnnotation(annotations map[string]string) (errs *apis.FieldError) {
 	for key := range annotations {
 		if strings.HasPrefix(key, autoscaling.GroupName) {
-			return apis.ErrGeneric(noAutoscalingMsg, apis.CurrentField)
+			return apis.ErrInvalidKeyName(key, apis.CurrentField, `autoscaling annotations must be put under "spec.template.metadata.annotations" to work`)
```

Comment:
well if we're not going to aggregate them with `Also`, then `errs` is not required.

Reply:
missed that, sorry :joy: